src/transport_ofi.c

```diff
@@ -518,26 +558,40 @@ int shmem_transport_init(long eager_size)
     shmem_free_list_init(sizeof(shmem_transport_ofi_long_frag_t),
                              init_long_frag);
 
-    allocate_fabric_resources(p_info, npes);
+    ret = allocate_fabric_resources(p_info, npes);
+    if(ret!=0)
+	return ret;
 
-    allocate_endpoints(p_info);
+    ret = allocate_endpoints(p_info);
+    if(ret!=0)
+	return ret;
 
-    allocate_cntr_and_cq();
+    ret = allocate_cntr_and_cq();
+    if(ret!=0)
+	return ret;
 
-    bind_resources_to_and_enable_ep();
+    ret = bind_resources_to_and_enable_ep();
+    if(ret!=0)
+	return ret;
 
-    allocate_recv_cntr_mr();
+    ret = allocate_recv_cntr_mr();
+    if(ret!=0)
+	return ret;
 
-    atomic_limitations_check();
+    ret = atomic_limitations_check();
+    if(ret!=0)
+	return ret;
 
-    exchange_and_av_insert(npes);
+    ret = exchange_and_av_insert(npes);
+    if(ret!=0)
+	return ret;
 
     fi_freeinfo(p_info);
 
     if(provname)
       p_info->fabric_attr->prov_name = NULL;
 
-    return 0;
+    return ret;
```

Comment:
Similar comment as above -- I think I prefer return 0 here, unless there's a way to reach this statement after an error has occurred.